modules/checkout/src/Plugin/Commerce/CheckoutPane/Login.php

```diff
@@ -243,8 +242,11 @@ public function validatePaneForm(array &$pane_form, FormStateInterface $form_sta
     $values = $form_state->getValue($pane_form['#parents']);
     $username = $values['returning_customer']['name'];
     $password = trim($values['returning_customer']['password']);
+    // Setting url for "forgotten password" link in the error message.
+    $route_parameters = !empty($username) ? ['name' => $username] : [];
```

Comment:
$route_parameters should be $query. The real route parameters are the empty array passed to Url.